src/wallet.py

```diff
@@ -0,0 +1,92 @@
+import json
+from json.decoder import JSONDecodeError
+
+import structlog
+from nacl.encoding import HexEncoder
+from nacl.signing import SigningKey
+import base64
+from cryptography.fernet import Fernet
+
+
+logger = structlog.get_logger(__name__)
+
+class wallet:
+    PRIVATE_KEY = ""
+    PUBLIC_KEY = ""
+    
+    def generate_wallet():
+        password = input("\nSe necesita una contraseña, escriba una: ")    
+        passws = password.encode('utf-8') #Se codifica para que lo admita
+        #Fernet pide una longitud de 32 bytes, a la contraseña que se usa
+        #se añaden tantos '0' hasta completarlo
+        while (len(passws) < 32):
+            passws = passws + b"0"
+            
+        pwd = base64.b64encode(passws)
+        f = Fernet(pwd)
+        private_key = SigningKey.generate()
+        public_key = private_key.verify_key
+        payload = {
+            "private_key": private_key.encode(encoder=HexEncoder).decode(),
+            "public_key": "<PERSONALIZABLE>" + public_key.encode(encoder=HexEncoder).decode(),
+        }
+        
+        with open("wallet.json", "w") as file:
```

Comment:
Can we make the path an input argument of the method? So its not hardcoded as `wallet.json`?

Reply:
I tried to do it this way from the beginning, but I couldn't do it. Besides, it gives you the opportunity to save it somewhere else if you want to.

Reply:
Thank you, it was so obvious I didn't even notice.

Reply:
Included